schemas/development/xarf_shared.schema.json

```diff
@@ -385,6 +451,44 @@
         }
       }
     },
+    "WithAttacker": {
+      "type": "object",
+      "description": "Specify the source of an attack. Used when attack source is not the subject of the report, eg. in an PotentiallyCompromisedAccount report.",
+      "properties": {
+        "AttackerIp": {
+          "description": "attacker ip",
+          "oneOf": [
+            {
+              "type": "string",
+              "format": "ipv4"
+            },
+            {
+              "type": "string",
+              "format": "ipv6"
+            }
+          ]
+        },
+        "AttackerPort": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "type": "integer",
+                "description": "attack port",
+                "minimum": 0,
+                "maximum": 65535
+              }
+            },
+            {
+              "type": "integer",
+              "description": "attack port",
+              "minimum": 0,
+              "maximum": 65535
+            }
+          ]
+        }
+      }
+    },
```

Comment:
I don't like the fact that this is essentially the same information as in `IpBasedReport`, but with different names. Can't we somehow combine these two?

Reply:
The `PotentiallyCompromisedAccount` report specifically doesn't allow `SourceIp`, `SoucePort` and `SourceUrl` and instead renames them to `AttackerIp` and `AttackerPort`. 

I've implemented it this way to prevent any confusion on who the subject of the report is. In this case, it's the user that has been attacked, not the attacker itself.

It could be possible to reuse the definition of what an `ip` and `port` are in the context of xarf, then reuse them for both `Source*` and `Attacker*`.

Reply:
Yes, as I think about this a bit more, using different names here makes sense.